### DISCUSSION TITLE
How to contribute ?

### DISCUSSION THREAD

**Discussion by marcpley** [pawelsalawa/sqlitestudio#5098] [General]
I recently place several enhancement demands. I ask myself to contribute on code developpement.
What is the smallest Qt 5.12 to work on source file ?
Qt Community Edition
Qt Design Studio Community Edition (DSCE)
Qt group website propose several options...

**Comment by pawelsalawa** [comment]
Community Edition is enough.

If you need assistance with dev environment setup, don't hesitate to ask in this thread. I will try to help as much as I'm able to.

**Reply by marcpley** [reply]
Thanks :-)
Do I  need to download source cote of Qt and compile them ? I didn't find anything else.

**Reply by pawelsalawa** [reply]
You need to download online installer: https://download.qt.io/official_releases/online_installers/
The online installer requires you to have Qt account (it's free to set it up). Offline installers are also available around the internet, but you would have to find and download each part separately and configure them together. There are few things you should install:
- Qt 5.15.x
- QtCreator
- Tools/mingw 8.10

All these you will find in the online installer and it will set up everything for you.

**Reply by marcpley** [reply]
I don't find Tools/mingw 8.10 in online installer.
![Screenshot_20241105_145825](https://github.com/user-attachments/assets/b75600d8-3fd6-417f-bce6-45abcf4a2753)

**Reply by marcpley** [reply]
I'm under Linux mint, I suppose I don't need this mingw...

**Comment by marcpley** [comment]
Qt 5.12 installed
SQLS 3.4.4 source code installed
I've opened SQLiteStudio3.pro in QtCreator ans configure it.
![codesourceinstall](https://github.com/user-attachments/assets/46f1b2b5-32f4-4bc3-ba66-5c22e5f6076d)
When I try to compile I got an error:
![Screenshot_20241105_172753](https://github.com/user-attachments/assets/f1d423dd-8ee8-4a23-8605-786718d579d2)
There is no file readline.h in the source code I've download.

**Reply by pawelsalawa** [reply]
Right. Under Linux no mingw is needed.

For the `readline.h` you need `readline-devel` or similar package. Under Ubuntu it's installed with `sudo apt-get install libreadline-dev`. I'm not sure if it's the same under Mint, but I guess you will know better.

**Reply by pawelsalawa** [reply]
And don't forget you need 2 `pro` files open in QtCreator. You also need `Plugins.pro` and you need to compile `SQLiteStudio` first, then `Plugins`.

**Reply by marcpley** [reply]
Thanks. readline problem seem to be solved.
With Plugins.pro I got this error : 
:-1: error: [/home/marc/Documents/Projets/greli.net/sqlitestudio/output/build/Makefile:384 : sub-ScriptingTcl-qmake_all] Erreur 3
I think there is a problem with my output/build dir. It must be the same for the two pro files ? See my directory tree (picture above). I probably misunderstood your explanation here: 
https://github.com/pawelsalawa/sqlitestudio/wiki/Instructions_for_compilation_under_Linux

**Reply by pawelsalawa** [reply]
Instructions from your link are just for compiling. If you want to set up a dev environment, you should follow: https://github.com/pawelsalawa/sqlitestudio/wiki/Instructions_for_dev_setup_under_Linux

Important things are on the "Project" page of QtCreator, especially the output path.
For SQLiteStudio3 project (assuming that `/home/user/projects/sqlitestudio` is your source code root and the `SQLiteStudio3.pro` is in `/home/user/projects/sqlitestudio/SQLiteStudio3/`) you should have:
![image](https://github.com/user-attachments/assets/3e1620ba-284d-49dd-8bfc-847e77b389b0)

For Plugins it should be:
![image](https://github.com/user-attachments/assets/23c7f683-c613-4d6c-abaf-9299bc309b62)

And it is a good idea to configure the "Run" page of SQLiteStudio3 page this way:
![image](https://github.com/user-attachments/assets/21abe976-8d86-459f-9968-1eeef9402408)

**Reply by marcpley** [reply]
Thanks for those very helpful screenshots.
I set the same still have a difference:
![Screenshot_20241106_073431](https://github.com/user-attachments/assets/9b6a9288-b972-4dd1-9dfc-47c8cd50ce03)
And this error persists when building Plugins:
:-1: error: [/home/marc/Documents/Projets/greli.net/sqlitestudio/output/Plugins/Makefile:384 : sub-ScriptingTcl-qmake_all] Erreur 3

**Reply by marcpley** [reply]
This error is solved.

**Reply by pawelsalawa** [reply]
For the Run configuration you currently have "sqlitestudiocli" selected, which is a Console (command line) interface of SQLiteStudio (still in experimental stage).

You usually want to run "sqlitestudio". You can change it either from the drop down "Run configuration" at the screen you pasted,, or simply on the left side:
![image](https://github.com/user-attachments/assets/6a9b850c-daed-47a0-b62a-d744a4da979c)

**Comment by pawelsalawa** [comment]
Also, this error:
```
error: [/home/marc/Documents/Projets/greli.net/sqlitestudio/output/build/Makefile:384 : sub-ScriptingTcl-qmake_all] Erreur 3
```
This is just a last message. What's import is what are few lines that are before it. There are details of what actually the error was.

**Reply by marcpley** [reply]
I don't find any lines before it, there is just this error.

**Reply by marcpley** [reply]
When opening Qt Creator and opening *.pro file, I got this error:
:-1: error: /bin/sh: 1: tclsh: not found
:-1: error: Project ERROR: Could not find tclsh executable. ScriptingTcl plugin requires it to find out all Tcl libraries and headers. Make tclsh available in PATH.
I've installed tcl package and now:
:-1: error: Could not find tclConfig.sh file. You can define its absolute path by qmake parameter: TCL_CONFIG=/path/to/tclConfig.sh
tclConfig.sh file is not on my system. May I create an empty file ?

**Reply by marcpley** [reply]
I've installed tcl-dev package and it seem to be ok.

**Comment by marcpley** [comment]
Now sqlitestudio3 is built with warnings but no error :-)

When building Plugins I have this error:
```
/home/marc/Documents/Projets/greli.net/sqlitestudio/Plugins/ScriptingPython/scriptingpython.h:4: error: Python.h: No such file or directory
In file included from ../../../Plugins/ScriptingPython/scriptingpython.cpp:1:
../../../Plugins/ScriptingPython/scriptingpython.h:4:10: fatal error: Python.h: No such file or directory
    4 | #include <Python.h>
```
The file Python.h is here:
/usr/include/python3.8
/usr/include/python2.7
I removed Python 2.7 in Kits manager.
And added "/usr/include/python3.8" to my PATH but the error persist.
![image](https://github.com/user-attachments/assets/11ef01c0-b0e4-4e7e-9786-c3c91b391958)

**Reply by pawelsalawa** [reply]
Try adding `DEFINES += "PYTHON_VERSION=3.8"` to qmake args of "Plugins" project:
![image](https://github.com/user-attachments/assets/007f0fd6-93cd-4c82-a1f5-01ca9fc098f7)

Do not create separate project config for Python. I mean this:
![image](https://github.com/user-attachments/assets/4ebcc123-3d7f-44d3-a8ea-6b410e9a7f09)
It should not be needed.

**Reply by marcpley** [reply]
I set all that.
Now I have:
(command line):1: error: Assignment needs exactly one word on the left hand side.

Is there a verbose mode to have more information ? 
Perhaps we can plan a 5 min visio where you could check my config ?

**Reply by pawelsalawa** [reply]
Sure we can connect. Regarding details please email me on support@sqlitestudio.pl and we will talk.
